xpcom/threads/Mutex.h

```diff
@@ -216,6 +217,23 @@ BaseAutoLock(MutexType&) -> BaseAutoLock<MutexType&>;
 typedef detail::BaseAutoLock<Mutex&> MutexAutoLock;
 typedef detail::BaseAutoLock<OffTheBooksMutex&> OffTheBooksMutexAutoLock;
 
+// Locks an ordered mutex. When events are disallowed on the current thread,
+// the lock will be unordered and could occur at a different point when replaying.
+class MutexAutoLockMaybeEventsDisallowed {
```

Comment:
Nit: For clarity, this name should likely be prefixed with `Ordered` to match the static variant.